File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Set<Character> seen = new HashSet<>();

        int r = 0;
        int l = 0;
        int longest = 1;

        if(s.isEmpty()){
            return 0;
        }

        while(r < s.length()){
            char ch = s.charAt(r);
            if(!(seen.contains(ch))){
                seen.add(ch);
                longest = Math.max(longest,r - l + 1);
                r++;
            }else{
                while(seen.contains(s.charAt(r))){
                    seen.remove(s.charAt(l));
                    l++;
                }
            }
        }

        return longest;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let seen = new Set(nums); 
        let longest = 0 ; 
        for(let num of seen){
            if(!seen.has(num - 1)){
                let length = 1 
                while(seen.has(num + length)){
                    length++
                }
                longest = Math.max(longest , length )
            }
        }return longest
    }
}
